src/debuginfo.cpp

```diff
@@ -811,14 +831,17 @@ void jl_getFunctionInfo(const char **name, size_t *line, const char **filename,
         if ((*it).second.func) {
             DISubprogram debugscope =
                 DISubprogram(prev.Loc.getScope((*it).second.func->getContext()));
-            *filename = debugscope.getFilename().data();
+            jl_copy_str(filename, debugscope.getFilename().data());
```

Comment:
> I'm not sure if these StringRefs would actually leak: they don't own the underlying buffer and AFAIK .data() shouldn't need to allocate. (but valgrind might not be able to figure that out?)

The point is exactly the opposite. We were very inconsistent about how to free `*filename` and `*name` (therefore in the old code, we simply never free them). There are places where we assign constant (or otherwise known to be live after the function returns) strings, and other places where we assign a malloc'd pointer. 

> Also, we should probably be more careful about copying from StringRef because they are not guaranteed to be null-terminated.

I didn't know that. However, this is not a new assumptions with this PR.